fvm/evm/emulator/state/base.go

```diff
@@ -448,30 +463,23 @@ func (v *BaseView) storeSlot(sk types.SlotAddress, data gethCommon.Hash) error {
 	return col.Set(sk.Key.Bytes(), data.Bytes())
 }
 
-func (v *BaseView) getSlotCollection(addr gethCommon.Address) (*Collection, error) {
-	acc, err := v.getAccount(addr)
-	if err != nil {
-		return nil, err
-	}
-	if acc == nil {
-		return nil, fmt.Errorf("slot belongs to a non-existing account")
-	}
-	if !acc.HasCode() {
-		return nil, fmt.Errorf("slot belongs to a non-smart contract account")
-	}
+func (v *BaseView) getSlotCollection(acc *Account) (*Collection, error) {
+	var err error
+
 	if len(acc.CollectionID) == 0 {
 		// create a new collection for slots
 		col, err := v.collectionProvider.NewCollection()
 		if err != nil {
 			return nil, err
 		}
 		// cache collection
-		v.slots[addr] = col
+		v.slots[acc.Address] = col
 		// update account's collection ID
 		acc.CollectionID = col.CollectionID()
 		err = v.storeAccount(acc)
```

Comment:
nit: here it's possible that collection will actually be assigned, but the error of storing account will be non-nil, in which case it will return a valid collection and an error, probably safer to only return error if one occurs and return nil instead of collection

Reply:
updated the logic